### PR TITLE
Copy-to-default-graph-generalization

### PR DESCRIPTION
**Pull request by glguy**
See #912

### CODE REVIEW

File: cli/rack/manifest.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
         'name': {'type': 'string'},
         'description': {'type': 'string'},
 
-        'copy-to-default-graph':            {'type': 'boolean'},
+        'copy-to-graph':                    {'type': 'string'},
         'perform-entity-resolution':        {'type': 'boolean'},
```
[the remark this answers is not in the document]
Comment (glguy) [reply cli/rack/manifest.py]: I guess now's as good a time as any

File: cli/rack/manifest.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
...
         'perform-triplestore-optimization': {'type': 'boolean'},
```
[the remark this answers is not in the document]
Comment (glguy) [reply cli/rack/manifest.py:18]: sure

File: cli/rack/__init__.py [hunk 3]
```diff
@@ -501,7 +500,7 @@ def build_manifest_driver(
     zipfile_path: Path
 ) -> None:
 
-    with tempfile.TemporaryDirectory() as outdir:
+    with TemporaryDirectory() as outdir:
```
Comment (Ptival) [inline cli/rack/__init__.py:503]: Is it the case that `builder.model_graphs` and `builder.data_graphs` depend on having created the archive?
At face value it looks like it's never explicitly used, but I assume there's a reason why it needs to be created, even if temporarily.

File: cli/rack/__init__.py [hunk 4]
```diff
@@ -541,51 +548,46 @@ def ingest_manifest_driver(
...
-
-        if manifest.getPerformOptimization():
-            invoke_optimization(optimization_url)
+            if not manifest.getNodegroupsFootprint() == []:
+                delete_nodegroups_driver(manifest.getNodegroupsFootprint(), True, True, True, base_url)
+
+        for (step_type, step_data) in manifest.steps:
+            if StepType.DATA == step_type:
+                stepFile = base_path / step_data
+                ingest_data_driver(stepFile, base_url, None, None, triple_store, triple_store_type, False)
```
Comment (Ptival) [inline cli/rack/__init__.py:557]: `None, None` here mean that both the model graphs and the data graphs are the default graph?

**Review by Ptival — APPROVED** [review]
Looks fine. Left two questions to understand a bit of the context.

File: cli/rack/__init__.py (hunk 4 above)
Comment (glguy) [reply cli/rack/__init__.py:557]: It means they aren't overridden from what's specified in their import.yaml files.

When you're using `rack data import` directly you can override those graph URLs with a CLI flag

File: cli/rack/__init__.py (hunk 3 above)
Comment (glguy) [reply cli/rack/__init__.py:503]: I'm currently only collecting those graph names as debugging output. I am imagining that we could extend that in the future to report and error if the URLs were missing from the manifest's footprint:

**Pull request merged**